contracts/multisig-prover/src/execute.rs

```diff
@@ -274,27 +277,13 @@ pub fn should_update_worker_set(
     cur_workers: &WorkerSet,
     max_diff: usize,
 ) -> bool {
-    let new_workers_signers = new_workers
-        .signers
-        .values()
-        .cloned()
-        .collect::<Vec<Signer>>();
-
-    let cur_workers_signers = cur_workers
-        .signers
-        .values()
-        .cloned()
-        .collect::<Vec<Signer>>();
-
-    new_workers_signers
-        .iter()
-        .filter(|item| !cur_workers_signers.contains(item))
-        .count()
-        + cur_workers_signers
-            .iter()
-            .filter(|item| !new_workers_signers.contains(item))
-            .count()
-        > max_diff
+    signers_difference_count(&new_workers.signers, &cur_workers.signers).saturating_add(
+        signers_difference_count(&cur_workers.signers, &new_workers.signers),
```

Comment:
extract this into `signers_symmetric_difference_count` to make it easier to understand the condition